src/_P092_DLbus.ino

```diff
@@ -0,0 +1,1180 @@
+        value = sensorvalue * 0.1;
+        break;
+      case RAYS:
+        value = sensorvalue;
+        break;
+      case VOLUME_FLOW:
+        value = sensorvalue * 4;
+        break;
+      case ROOM:
+        ReadData.mode = (sensorvalue & 0x600) >> 9;
+        value = (sensorvalue & 0x1ff) * 0.1;
+        break;
+      default:
+        return false;
+    }
+  }
+  else { // sign negative
+    sensorvalue |= 0xf000;
+    // calculations for different sensor types
+    switch (sensortype) {
+      case DIGITAL:
+        value = true;
+        break;
+      case TEMP:
+        value = (sensorvalue - 65536) * 0.1;
+        break;
+      case RAYS:
+        value = sensorvalue - 65536;
+        break;
+      case VOLUME_FLOW:
+        value = (sensorvalue - 65536) * 4;
+        break;
+      case ROOM:
+        ReadData.mode = (sensorvalue & 0x600) >> 9;
+        value = ((sensorvalue & 0x1ff) - 65536) * 0.1;
+        break;
+      default:
+        return false;
+    }
+  }
+  ReadData.value = value;
+  return true;
+}
+
+boolean P092_fetch_output(int number) {
+  int outputs;
+
+  if (DataSettings.Outputuint8_ts > 1)
+    outputs = (uint8_tStream[DataSettings.IdxOutput + 1] << 8) | uint8_tStream[DataSettings.IdxOutput];
+  else
+    outputs = uint8_tStream[DataSettings.IdxOutput];
+  ReadData.value = (!!(outputs & (1 << (number - 1))));
+  return true;
+}
+
+boolean P092_fetch_speed(int number) {
+  uint8_t speeduint8_t;
+
+  speeduint8_t = uint8_tStream[DataSettings.IdxDrehzahl + (number - 1)];
+  if ((speeduint8_t & 0x80) == 0x80)
+    return false;
+  ReadData.value = (speeduint8_t & 0x1f);
+  return true;
+}
+
+boolean P092_fetch_analog(int number) {
+  uint8_t analoguint8_t;
+
+  analoguint8_t = uint8_tStream[DataSettings.IdxAnalog + (number - 1)];
+  if ((analoguint8_t & 0x80) == 0x80)
+    return false;
+  ReadData.value = (analoguint8_t * 0.1);
+  return true;
+}
+
+boolean P092_CheckHmRegister(int number) {
+  switch (number) {
+    case 1:
+      if ((uint8_tStream[DataSettings.IdxHmRegister] & 0x1) != 0x01)
+        return false;
+      power_index = DataSettings.IdxHeatMeter1;
+      kwh_index = DataSettings.IdxkWh1;
+      mwh_index = DataSettings.IdxMWh1;
+      break;
+    case 2:
+      if ((uint8_tStream[DataSettings.IdxHmRegister] & 0x2) != 0x02)
+        return false;
+      power_index = DataSettings.IdxHeatMeter2;
+      kwh_index = DataSettings.IdxkWh2;
+      mwh_index = DataSettings.IdxMWh2;
+      break;
+    case 3:
+      if ((uint8_tStream[DataSettings.IdxHmRegister] & 0x4) != 0x04)
+        return false;
+      power_index = DataSettings.IdxHeatMeter3;
+      kwh_index = DataSettings.IdxkWh3;
+      mwh_index = DataSettings.IdxMWh3;
+      break;
+    default:
+      return false;
+  }
+  return true;
+}
+
+boolean P092_fetch_heatpower(int number) {
+  // current power
+  int32_t high;
+
+  if (P092_CheckHmRegister(number) == false)
+    return false;
+  uint8_t b1 = uint8_tStream[power_index];
```

Comment:
`power_index` is not initialized when constructed and I am not sure it is set when this function is first called.

Reply:
Done!